packages/core/src/tracing/span.ts

```diff
@@ -602,6 +604,16 @@ export class Span implements SpanInterface {
     return spanToTraceContext(this);
   }
 
+  /** @inheritdoc */
+  public getMetricSummary(): MetricSummaryAggregator | undefined {
+    return this._metricSummary;
```

Comment:
We'll end up having to change this for OTEL spans. To get ahead of that what do you think about something like:

```ts
// example
const metricSummary = getMetricSummaryFromSpan(span);

// later
const SPAN_METRIC_SUMMARY = new WeakMap<Span, MetricSummaryAggregator>()

function getMetricSummaryFromSpan(span: Span): MetricSummaryAggregator | undefined {
  return SPAN_METRIC_SUMMARY.get(span);
}
```

This way spans and transactions can just call `getMetricSummaryFromSpan` themselves, and we don't have to worry about polluting the public API.

Reply:
Yep that should work. The only downside I can see with this approach is that we'll end up with the Metric summary fetch from weakmap + `getJson()` implementation in the default browser bundle. 

That's not the end of the world since it's not that complicated!

Reply:
> we'll end up with the Metric summary fetch from weakmap + getJson() implementation in the default browser bundle

That is fine on my end, we can introduce tree shaking flags if we really need to get rid of it.